recommendation/microsoft.owin.json

```diff
@@ -0,0 +1,352 @@
+                "NotContains": "usingMicrosoft.Owin"
+              }
+            },
+            {
+              "Name": "RemoveDirective",
+              "Type": "Using",
+              "Value": "Owin",
+              "Description": "Remove Owin namespace",
+              "ActionValidation": {
+                "Contains": "",
+                "NotContains": "usingOwin"
+              }
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "Type": "Class",
+      "Name": "IOwinContext",
+      "Value": "Microsoft.Owin.IOwinContext",
+      "KeyType": "Identifier",
+      "ContainingType": "",
+      "RecommendedActions": [
+        {
+          "Source": "Amazon",
+          "Preferred": "Yes",
+          "TargetFrameworks": [
+            {
+              "Name": "netcoreapp3.1",
+              "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
+            },
+            {
+              "Name": "net5.0",
+              "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
+            }
+          ],
+          "Description": "Replace IOwinContext with HttpContext\r\nAdd Microsoft.AspNetCore.Http.HttpContext namespace\r\nRemove Microsoft.Owin.IOwinContext\r\n",
+          "Actions": [
+            {
+              "Name": "ReplaceIdentifier",
+              "Type": "Identifier",
+              "Value": "HttpContext",
+              "Description": "Replace IOwinContext with HttpContext",
+              "ActionValidation": {
+                "Contains": "HttpContext",
+                "NotContains": "IOwinContext"
+              }
+            },
+            {
+              "Name": "AddDirective",
+              "Type": "Using",
+              "Value": "Microsoft.AspNetCore.Http",
+              "Description": "Add Microsoft.AspNetCore.Http namespace",
+              "ActionValidation": {
+                "Contains": "usingMicrosoft.AspNetCore.Http",
+                "NotContains": ""
+              }
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "Type": "Class",
+      "Name": "OwinMiddleware",
+      "Value": "Microsoft.Owin.OwinMiddleware",
+      "KeyType": "BaseClass",
+      "ContainingType": "",
+      "RecommendedActions": [
+        {
+          "Source": "Amazon",
+          "Preferred": "Yes",
+          "TargetFrameworks": [
+            {
+              "Name": "netcoreapp3.1",
+              "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
+            },
+            {
+              "Name": "net5.0",
+              "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
+            }
+          ],
+          "Description": "Removes base class of OwinMiddleware\r\n",
+          "Actions": [
+            {
+              "Name": "RemoveBaseClass",
+              "Type": "Class",
+              "Value": "OwinMiddleware",
+              "Description": "Removes base class OwinMiddleware",
+              "ActionValidation": {
+                "Contains": "",
+                "NotContains": "OwinMiddleware"
```

Comment:
Are we adding a different baseclass? I thought this caused the baseclass to be removed, but not the ":"

Reply:
I changed it to a comment since currently you are right it doesn't remove the whole base class. The comment explains the same, that it must be removed.